### PR TITLE
Includes support for specifying MAC address when creating VM.

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY

Includes support for specifying MAC address when creating VM.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

vsphere_guest
##### ANSIBLE VERSION

```
ansible 2.4.0
  config file = 
  configured module search path = [u'/home/***/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /***/virtualenv/lib/python2.7/site-packages/ansible
  executable location = /***/virtualenv/bin/ansible
  python version = 2.7.13 (default, Jan 12 2017, 17:59:37) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

```

##### ADDITIONAL INFORMATION

Implementing @patbaker82 's PR #3643
https://github.com/ansible/ansible-modules-core/pull/3643

### CODE REVIEW

**Comment by gundalow** [comment]
```
lib/ansible/modules/cloud/vmware/vsphere_guest.py:1121:40: E712 comparison to False should be 'if cond is not False:' or 'if cond:'
lib/ansible/modules/cloud/vmware/vsphere_guest.py:1144:40: E712 comparison to False should be 'if cond is not False:' or 'if cond:'
lib/ansible/modules/cloud/vmware/vsphere_guest.py:1179:27: E712 comparison to False should be 'if cond is not False:' or 'if cond:'
```

**Comment by enasisnetwork** [comment]
Is there anything I'm supposed to do?

**Comment by gundalow** [comment]
Closing and reopening to trigger CI

File: lib/ansible/modules/cloud/vmware/vsphere_guest.py [hunk 1]
```diff
@@ -519,7 +521,11 @@ def add_nic(module, s, nfmor, config, devices, nic_type="vmxnet3", network_name=
             msg="Error adding nic backing to vm spec. No network type of:"
             " %s" % (network_type))
 
-    nic_ctlr.set_element_addressType("generated")
+    if mac_address == "GENERATED":
```
Comment (dagwieers) [inline lib/ansible/modules/cloud/vmware/vsphere_guest.py:524]: I don't like that this is uppercase. Why don't we set it to generated by default if it wasn't provided. (So set it to None by default).

Comment (dagwieers) [reply lib/ansible/modules/cloud/vmware/vsphere_guest.py:524]: This is also not documented in the module. If you are adding functionality and don't document it, what's the use ?

**Review by dagwieers — CHANGES_REQUESTED** [review]
Looking mostly good. A few things to clarify and more importantly, missing documentation for the new functionality.

**Comment by Akasurde** [comment]
@ghost Are you still working on this ? Let me know if not I can take over this.

**Comment by Akasurde** [comment]
@ghost Are you working on this change ?

**Comment by dagwieers** [comment]
@Akasurde It seems that user was deleted from Github...

**Comment by Akasurde** [comment]
Original author deleted from GitHub. Closing this PR.

**Pull request closed without merging**